src/journal/journal-file.c

```diff
@@ -1862,7 +1860,15 @@ static int journal_file_link_entry(JournalFile *f, Object *o, uint64_t offset) {
         /* Link up the items */
         n = journal_file_entry_n_items(o);
         for (uint64_t i = 0; i < n; i++) {
-                r = journal_file_link_entry_item(f, o, offset, i);
+                Object *item;
+                uint64_t p;
+
+                p = le64toh(o->entry.items[i].object_offset);
+                r = journal_file_move_to_object(f, OBJECT_DATA, p, &item);
+                if (r < 0)
+                        return r;
+
+                r = journal_file_link_entry_item(f, item, offset);
```

Comment:
hmm, what's the point of this change? you just shift a few lines from a so far nicely encapsulated call up?

Reply:
To make `journal_file_link_entry_item()` dealing with data objects only and avoid this function to deal with different type of objects which were all stored in `*o` variable.